### PR TITLE
Refactor remaining broker startup steps

### LINKED ISSUE #7539: Refactor broker bootstrap steps
**Description**

Take the existing bootstrap steps and refactor them so that they implement the `StartupStep` interface.

Depends on #7536, #7538 

**Details**
* Refactor each method called in `Broker.initStart()` to a dedicated step extending `AbstractStartupStep`
* While doing that define a `BrokerBootstrapContext` interface and default implementation class
* Use `LifecycleManager` in `Broker` to trigger and monitor lifecycle changes
* Investigate how to connect `LifecycleManager` to monitoring components. If it's trivial, implement it right away; otherwise create a dedicated issue.

### PR DESCRIPTION
**Pull request by pihme**
## Description

Bulk PR with the remaining work for #7539 

## Related issues

closes #7539 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [X] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/test/java/io/camunda/zeebe/broker/bootstrap/ClusterServicesStepTest.java [hunk 1]
```diff
@@ -50,14 +53,15 @@ void shouldCompleteFutureOnStartup() {
     sut.startupInternal(mockBrokerStartupContext, CONCURRENCY_CONTROL, future);
 
     // then
-    assertThat(future.isDone()).isTrue();
-    assertThat(future.isCompletedExceptionally()).isFalse();
+    assertThat(future).succeedsWithin(TIME_OUT);
```
Comment (npepinpe) [inline broker/src/test/java/io/camunda/zeebe/broker/bootstrap/ClusterServicesStepTest.java:56]: :wrench: You can access the result after a `succeedsWithin`, e.g. `assertThat(future).succeedsWithin(TIME_OUT).isNotNull()`. I use [this IntelliJ extension](https://plugins.jetbrains.com/plugin/12195-concise-assertj-optimizing-nitpicker-cajon-) to learn these things as it suggests to me to combine assertions, or use more explicit ones. It's not always perfect, but it's generally pretty good.

File: broker/src/main/java/io/camunda/zeebe/broker/Broker.java [hunk 2]
```diff
@@ -82,16 +77,14 @@ public Broker(final SystemContext systemContext, final SpringBrokerBridge spring
...
-            healthCheckService);
+            healthCheckService,
+            buildExporterRepository(getConfig()),
+            additionalPartitionListeners);
 
     brokerStartupActor = new BrokerStartupActor(startupContext);
     scheduler.submitActor(brokerStartupActor);
   }
 
-  public void addPartitionListener(final PartitionListener listener) {
```
Comment (npepinpe) [inline broker/src/main/java/io/camunda/zeebe/broker/Broker.java:91]: :+1: I like having a fixed set of listeners from the start

**Review by npepinpe — APPROVED** [review]
:+1: 

Looks good, nothing to say really. I like that we now create the broker with a fixed set of listeners and removed the dynamic registration.

**Comment by pihme** [comment]
bors merge

**Pull request merged**